common/collection/sync_map.go

```diff
@@ -47,6 +47,24 @@ func (m *SyncMap[K, V]) Get(key K) (value V, ok bool) {
 	return
 }
 
+func (m *SyncMap[K, V]) GetOrSet(key K, value V) (v V, exist bool) {
```

Comment:
not part of this PR, but can you change the `*sync.RWMutex` to `sync.RWMutex` (no pointer)?

Reply:
updated